packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -428,8 +428,11 @@ function commitAllHostEffects() {
 
 function commitBeforeMutationLifecycles() {
   while (nextEffect !== null) {
-    const effectTag = nextEffect.effectTag;
+    if (__DEV__) {
+      ReactCurrentFiber.setCurrentFiber(nextEffect);
+    }
```

Comment:
No, I meant to put it outside (just like we do in other effect traversal). It doesn't really give us anything to reset it on every iteration if we're going to set it again at the very beginning of every iteration.

Reply:
Actually sorry, I misunderstood your question (I thought you meant a different line).

But here, too, it was intentional, to match what we do in the other effect traversal. I put it inside the loop because I want to set the current fiber to the current effect as we iterate through effects. Putting it outside would defeat the purpose. Each effect is a different fiber.